internal/super/generate.go

```diff
@@ -0,0 +1,192 @@
+	"github.com/spf13/cobra"
+)
+
+type generateFlagsDef struct {
+	Directory string `default:"" flag:"dir" info:"Directory to generate files in"`
+}
+
+var generateFlags = generateFlagsDef{}
+
+var GenerateCommand = &cobra.Command{
+	Use:     "generate",
+	Short:   "Generate new boilerplate files",
+	GroupID: "super",
+}
+
+var GenerateContractCommand = &command.Command{
+	Cmd: &cobra.Command{
+		Use:     "contract <name>",
+		Short:   "Generate a new contract",
+		Example: "flow generate contract HelloWorld",
+		Args:    cobra.ExactArgs(1),
+	},
+	Flags: &generateFlags,
+	RunS:  generateContract,
+}
+
+var GenerateTransactionCommand = &command.Command{
+	Cmd: &cobra.Command{
+		Use:     "transaction <name>",
+		Short:   "Generate a new transaction",
+		Example: "flow generate transaction SomeTransaction",
+		Args:    cobra.ExactArgs(1),
+	},
+	Flags: &generateFlags,
+	RunS:  generateTransaction,
+}
+
+var GenerateScriptCommand = &command.Command{
+	Cmd: &cobra.Command{
+		Use:     "script <name>",
+		Short:   "Generate a new script",
+		Example: "flow generate script SomeScript",
+		Args:    cobra.ExactArgs(1),
+	},
+	Flags: &generateFlags,
+	RunS:  generateScript,
+}
+
+func init() {
+	GenerateContractCommand.AddToParent(GenerateCommand)
+	GenerateTransactionCommand.AddToParent(GenerateCommand)
+	GenerateScriptCommand.AddToParent(GenerateCommand)
+}
+
+func generateContract(
+	args []string,
+	_ command.GlobalFlags,
+	logger output.Logger,
+	flow flowkit.Services,
+	state *flowkit.State,
+) (result command.Result, err error) {
+	return generateNew(args, "contract", logger)
+}
+
+func generateTransaction(
+	args []string,
+	_ command.GlobalFlags,
+	logger output.Logger,
+	flow flowkit.Services,
+	state *flowkit.State,
+) (result command.Result, err error) {
+	return generateNew(args, "transaction", logger)
+}
+
+func generateScript(
+	args []string,
+	_ command.GlobalFlags,
+	logger output.Logger,
+	flow flowkit.Services,
+	state *flowkit.State,
+) (result command.Result, err error) {
+	return generateNew(args, "script", logger)
+}
+
+func generateNew(
+	args []string,
+	templateType string,
+	logger output.Logger,
+) (result command.Result, err error) {
+	if len(args) < 1 {
+		return nil, fmt.Errorf("invalid number of arguments")
+	}
+
+	name := args[0]
+	var filename string
+
+	// Don't add .cdc extension if it's already there
+	if strings.HasSuffix(name, ".cdc") {
+		filename = name
+	} else {
+		filename = fmt.Sprintf("%s.cdc", name)
+	}
+
+	var fileToWrite string
+	var basePath string
+
+	if generateFlags.Directory != "" {
+		basePath = generateFlags.Directory
+	} else {
+		switch templateType {
+		case "contract":
+			basePath = "cadence/contracts"
+		case "script":
+			basePath = "cadence/scripts"
+		case "transaction":
+			basePath = "cadence/transactions"
+		default:
+			return nil, fmt.Errorf("invalid template type: %s", templateType)
+		}
+	}
+
+	switch templateType {
+	case "contract":
+		fileToWrite = fmt.Sprintf(`
+pub contract %s {
```

Comment:
```suggestion
access(all) contract %s {
```
Should we preemptively Cadence1.0-proof this?